.jenkins/cscs/env-gcc-cuda.sh

```diff
@@ -5,14 +5,14 @@
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 export CRAYPE_LINK_TYPE=dynamic
-export CXX_STD="14"
+export CXX_STD="17"
 
 module load daint-gpu
 module switch PrgEnv-cray PrgEnv-gnu
 module load cudatoolkit
 module load Boost/1.75.0-CrayGNU-20.11
 module load hwloc/.2.0.3
-spack load cmake@3.17.3
+spack load cmake@3.18.6 # Necessary to handle cuda_std_17 compiler feature
```

Comment:
Do we need the comment? Should we not rather just require `3.18`?

Reply:
Was planning to do it but in another PR, cause this was originally just c++17 and then I added the `enable_language(CUDA)` refactoring (as it was necessary), I think doing this in this PR is a bit too much, what do you think?

Reply:
Up to you. It should be done as soon as possible, but a separate PR is also fine by me.